Assets/Scripts/Enemy/EnemyController.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class EnemyController : MonoBehaviour
+{
+    private float elapsedTime;
+    public float distanceToPlayer;
+    private GameObject player;
+    public NavMeshAgent agent;
+    public Vector3 destPos;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        player = GameObject.FindGameObjectWithTag("Player");
+        agent = gameObject.GetComponent<NavMeshAgent>();
```

Comment:
```suggestion
        agent = GetComponent<NavMeshAgent>();
```
`gameObject.` is niet nodig. Dan verwijs je namelijk naar hetzelfde object als waar je nu in zit.

Als je boven de klasse de flag `[RequireComponent(typeof(NavMeshAgent))]` zet, weet je zeker dat het script dat component heeft. Als hij nog niet bestaat, wordt hij automatisch toegevoegd